src/sysfiles.h

```diff
@@ -401,6 +401,28 @@ extern void SySetErrorNo ( void );
 *F * * * * * * * * * * * * * file and execution * * * * * * * * * * * * * * *
 */
 
+typedef enum ForkState { PreFork, PostForkChild, PostForkParent } ForkState;
+typedef void (*ForkObserver)(ForkState);
+
+/*
+ *F RegisterSyForkObserver( <func> )
+ ** Register a function to be called before and after fork is called.
```

Comment:
So, when adding documentation comments, I'd either make them match the existing ones, or radically different; but this is kind of a mishmash (only one space after `**`, no empty line after the `*F` line, no start line with `/*****`.

If you don't mind, I can amend this PR accordingly.

Reply:
I'm happy to adjust it. I want to look at why hpc-gap is crashing (I suspect switching vfork for fork, which suprises me, as I expected that to make hpc-gap more stable, not less.